docs/workflows/common-workflow-steps/custom-script.md

```diff
@@ -395,4 +395,72 @@ This script generates a timestamped log file (e.g., `ac-log-2024-10-01-14-55.txt
 
 Ensure that the **Custom Script** step runs after the [**Export Build Artifacts**](/workflows/common-workflow-steps/export-build-artifacts) step to capture the full log.
 
-:::
\ No newline at end of file
+:::
+
+### How can I print the status of workflow steps with detailed information?
+
+If you want to track or share the status of your workflow or individual steps during a build, you can use the following environment variables:  
+
+- **`$AC_BUILD_STATUS`**: Displays the overall status of the workflow.
+- **`$AC_BUILD_STEP_STATUS`**: Displays detailed information about each executed step.  
+
+However, the output of `$AC_BUILD_STEP_STATUS` is in raw JSON format, which may not be easy to read directly. To make it more readable, you can use the following Ruby script to format and print the information in a user-friendly way:
+
+```ruby
+require 'json'
+
+puts "AC_BUILD_STATUS: #{ENV['AC_BUILD_STATUS']}"
+# Read the environment variable
+json_data = ENV['AC_BUILD_STEPS_STATUS']
+
+begin
+  # Parse and beautify the JSON
+  parsed_data = JSON.parse(json_data)
+  pretty_json = JSON.pretty_generate(parsed_data)
+  
+  # Output the formatted JSON
+  puts "AC_BUILD_STEPS_STATUS:"
+  puts pretty_json
+rescue JSON::ParserError => e
+  puts "Failed to parse JSON: #{e.message}"
+end
+```
```

Comment:
When we use this sample script as the first step in the workflow. Is there an edge case there that should be handled in the script or mentioned as an info box?

Reply:
When I run it on the first line, the output is as follows:
```yaml
AC_BUILD_STATUS: Success
AC_BUILD_STEPS_STATUS:
[
]
```

 I already wrote in the `$AC_BUILD_STEPS_STATUS` desc that we are displaying the results of the **executed** steps. And I added the "... so far" statement to the `$AC_BUILD_STATUS` desc (it was already added on the reserved env page) so I think it is not necessary. 

@csonuryilmaz Is these desc enough for you too?

Reply:
OK, there is nothing to do extra here.